src/core/ext/filters/client_channel/lb_policy.h

```diff
@@ -261,8 +260,7 @@ class LoadBalancingPolicy : public InternallyRefCounted<LoadBalancingPolicy> {
   /// Note that the LB policy gets the set of addresses from the
   /// GRPC_ARG_SERVER_ADDRESS_LIST channel arg.
   virtual void UpdateLocked(const grpc_channel_args& args,
-                            RefCountedPtr<Config> lb_config) {
-    std::move(lb_config);  // Suppress clang-tidy complaint.
+                            RefCountedPtr<Config> lb_config) {  // NOLINT
```

Comment:
Why is `NOLINT` needed here, but not in the other locations?  Is it because the parameter has a name here?  If so, maybe just remove the parameter name and the `NOLINT` comment?

Reply:
This is to suppress clang-tidy's [performance-unnecessary-value-param](https://clang.llvm.org/extra/clang-tidy/checks/performance-unnecessary-value-param.html) warning.
The copy of RefCountedPtr looks expensive to clang-tidy, and it recommends us to pass it as a const ref. Though, we can't follow it as it's a virtual method.

Reply:
Isn't this exactly the same case as the other sites you're changing?  The `std::move()` call was added to work around this exact warning.  I removing the parameter name makes the warning go away in all of these other cases, why isn't that sufficient here?  Why is the `NOLINT` needed in this case but not the others?

Reply:
The [implementation of performance-unnecessary-value-param](https://github.com/llvm-mirror/clang-tools-extra/blob/0b9bba1a26c3862a0d6ac24bba7fb8b1a15f5c80/clang-tidy/performance/UnnecessaryValueParamCheck.cpp#L80) warns only when the parameter [isExpensiveToCopy](https://github.com/llvm-mirror/clang-tools-extra/blob/master/clang-tidy/utils/TypeTraits.cpp#L41). If the parameter has deleted copy ctor, it's not expensive to copy for clang-tidy.

IIUC, LoadBalancingPolicy::Args and UniquePtr didn't hit the check because their copy ctor is deleted.